ibllib/pipes/ephys_preprocessing.py

```diff
@@ -864,14 +865,153 @@ def get_signatures(self, **kwargs):
+        'output_files': [
+            ('_ibl_leftCamera.dlc.pqt', 'alf', True),
+            ('_ibl_rightCamera.dlc.pqt', 'alf', True),
+            ('_ibl_bodyCamera.dlc.pqt', 'alf', True),
+            ('leftCamera.ROIMotionEnergy.npy', 'alf', True),
+            ('rightCamera.ROIMotionEnergy.npy', 'alf', True),
+            ('bodyCamera.ROIMotionEnergy.npy', 'alf', True),
+            ('leftROIMotionEnergy.position.npy', 'alf', True),
+            ('rightROIMotionEnergy.position.npy', 'alf', True),
+            ('bodyROIMotionEnergy.position.npy', 'alf', True),
+        ],
+    }
+
+    def _check_dlcenv(self):
+        """Check that scripts are present, dlcenv can be activated and get iblvideo version"""
+        assert len(list(self.scripts.rglob('run_dlc.*'))) == 2, \
+            f'Scripts run_dlc.sh and run_dlc.py do not exist in {self.scripts}'
+        assert len(list(self.scripts.rglob('run_motion.*'))) == 2, \
+            f'Scripts run_motion.sh and run_motion.py do not exist in {self.scripts}'
+        assert self.dlcenv.exists(), f"DLC environment does not exist in assumed location {self.dlcenv}"
+        command2run = f"source {self.dlcenv}; python -c 'import iblvideo; print(iblvideo.__version__)'"
+        process = subprocess.Popen(
+            command2run,
+            shell=True,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            executable="/bin/bash"
+        )
+        info, error = process.communicate()
+        if process.returncode != 0:
+            raise AssertionError(f"DLC environment check failed\n{error.decode('utf-8')}")
+        version = info.decode("utf-8").strip().split('\n')[-1]
+        return version
+
+    @staticmethod
+    def _video_intact(file_mp4):
+        """Checks that the downloaded video can be opened and is not empty"""
+        cap = cv2.VideoCapture(str(file_mp4))
+        frame_count = cap.get(cv2.CAP_PROP_FRAME_COUNT)
+        intact = True if frame_count > 0 else False
+        cap.release()
+        return intact
+
+    def _run(self, cams=None, overwrite=False):
+        # Default to all three cams
+        cams = cams or ['left', 'right', 'body']
+        cams = [assert_valid_label(cam) for cam in cams]
+        # Set up
+        self.session_id = self.one.path2eid(self.session_path)
+        actual_outputs = []
+
+        # Loop through cams
+        for cam in cams:
+            # Catch exceptions so that following cameras can still run
+            try:
+                # If all results exist and overwrite is False, skip computation
+                expected_outputs_present, expected_outputs = self.assert_expected(self.output_files, silent=True)
+                if overwrite is False and expected_outputs_present is True:
+                    actual_outputs.extend(expected_outputs)
+                    continue
+                else:
+                    file_mp4 = self.session_path.joinpath('raw_video_data', f'_iblrig_{cam}Camera.raw.mp4')
```

Comment:
Better to glob i.e. `next(self.session_path.joinpath('raw_video_data').glob(f'_iblrig_{cam}Camera.raw*mp4')`